providers/dns/brandit/brandit.go

```diff
@@ -0,0 +1,181 @@
+	config.APIKey = values[EnvAPIKey]
+	config.APIUsername = values[EnvAPIUsername]
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for BrandIT.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		return nil, errors.New("brandit: the configuration of the DNS provider is nil")
+	}
+
+	client, err := internal.NewClient(config.APIUsername, config.APIKey)
+	if err != nil {
+		return nil, fmt.Errorf("brandit: %w", err)
+	}
+
+	if config.HTTPClient != nil {
+		client.HTTPClient = config.HTTPClient
+	}
+
+	return &DNSProvider{
+		config:  config,
+		client:  client,
+		records: make(map[string]string),
+	}, nil
+}
+
+// Present creates a TXT record using the specified parameters.
+func (d *DNSProvider) Present(domain, token, keyAuth string) error {
+	info := dns01.GetChallengeInfo(domain, keyAuth)
+
+	authZone, err := dns01.FindZoneByFqdn(info.EffectiveFQDN)
+	if err != nil {
+		return fmt.Errorf("brandit: %w", err)
+	}
+
+	subDomain, err := dns01.ExtractSubDomain(info.EffectiveFQDN, authZone)
+	if err != nil {
+		return fmt.Errorf("brandit: %w", err)
+	}
+
+	record := internal.Record{
+		Type:    "TXT",
+		Name:    subDomain,
+		Content: info.Value,
+		TTL:     d.config.TTL,
+	}
+
+	resp, err := d.client.PresentRecord(dns01.UnFqdn(authZone), record)
+	if err != nil {
+		return fmt.Errorf("brandit: add record: %w", err)
+	}
+
+	if resp.Status == internal.StatusSuccess {
+		d.recordsMu.Lock()
+		d.records[token] = resp.Record
+		d.recordsMu.Unlock()
+
+		return nil
+	}
+
+	return nil
+}
+
+// CleanUp removes the TXT record matching the specified parameters.
+func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
+	info := dns01.GetChallengeInfo(domain, keyAuth)
+
+	authZone, err := dns01.FindZoneByFqdn(info.EffectiveFQDN)
+	if err != nil {
+		return fmt.Errorf("brandit: %w", err)
+	}
+
+	// gets the record's unique ID
+	d.recordsMu.Lock()
+	DNSRecord, ok := d.records[token]
+	d.recordsMu.Unlock()
+	if !ok {
+		return fmt.Errorf("brandit: unknown record ID for '%s' '%s'", info.EffectiveFQDN, token)
+	}
+
+	resp, err := d.client.CleanUpRecord(dns01.UnFqdn(authZone), DNSRecord)
+	if err != nil {
+		return fmt.Errorf("brandit: delete record: %w", err)
+	}
+
+	// deletes record ID from map
+	d.recordsMu.Lock()
+	delete(d.records, token)
+	d.recordsMu.Unlock()
+
+	if resp.Status == internal.StatusSuccess {
+		return nil
+	}
+
+	return nil
+}
+
+// Timeout returns the timeout and interval to use when checking for DNS propagation.
+// Adjusting here to cope with spikes in propagation times.
+func (d *DNSProvider) Timeout() (timeout, interval time.Duration) {
+	return d.config.PropagationTimeout, d.config.PollingInterval
+}
+
+// Sequential All DNS challenges for this provider will be resolved sequentially.
+// Returns the interval between each iteration.
+func (d *DNSProvider) Sequential() time.Duration {
+	return d.config.SequenceInterval
+}
```

Comment:
Why do you use the sequential approach?
does the provider not support having multiple DNS records for 1 domain?

Reply:
We missed that, i just tested and everything works fine